src/devices/Cargo.toml

```diff
@@ -18,3 +18,6 @@ versionize = { version = "0.1.1" }
 versionize_derive = { git = "https://github.com/firecracker-microvm/versionize_derive", tag = "v0.1.0" }
 virtio_gen = { path = "../virtio_gen" }
 mmds = { path = "../mmds" }
+kvm-ioctls = { git = "https://github.com/firecracker-microvm/kvm-ioctls", tag = "v0.5.0-2" }
+timer = ">=0.1.3"
+chrono = ">=0.2"
```

Comment:
I think we need a different mechanism than relying on `timer` (which also brings in `chrono`) for multiple reasons:
 - `timer` crate has an incompatible `MPL-2.0` license
 - `timer` is implemented using a multi-thread timer model: spawns a new thread to do the work after a certain time. This is not in line with our security model/posture where we want to keep strict control over process threads (our `seccomp` filters do not allow spawning new threads).
 - `chrono` is a bloated dependency with a long-ish trail of child dependencies. We'd like to avoid it since the use-case for it is simple enough to cover with `std::time`.

Possible alternative: use `TimerFd` - for example that's what we use in our Rate Limiter implementation: https://github.com/firecracker-microvm/firecracker/blob/master/src/rate_limiter/src/lib.rs

Reply:
thanks. I'll try to replace them with `TimerFd`. ;)

Reply:
Our device model uses a single emulation thread that asynchronously handles device events. `TimerFd` would be such a `gpio_pl061` device event.

The device needs to use our `EventManager` [Subscriber interface](https://github.com/firecracker-microvm/firecracker/blob/db84b948461bddcc89c0b69aa5639368a7d7c3ef/src/polly/src/event_manager.rs#L50) to register async events.
A simple example device with a single event can be seen in our `Serial device` implementation: https://github.com/firecracker-microvm/firecracker/blob/db84b948461bddcc89c0b69aa5639368a7d7c3ef/src/devices/src/legacy/serial.rs#L290